src/Storages/StorageFile.cpp

```diff
@@ -513,6 +528,22 @@ class StorageFileSource : public ISource
         , context(context_)
         , max_block_size(max_block_size_)
     {
+        /// proton : starts
+        is_streaming = is_streaming_;
+        if (storage->isColumnOriented())
+            block_for_format = storage_snapshot->getSampleBlockForColumns(columns_description.getNamesOfPhysical());
+        else
+            block_for_format = storage_snapshot->metadata->getSampleBlock();
+
+        if (is_streaming)
+        {
+            for (const auto * reserved_col :
+                 {&ProtonConsts::STREAMING_WINDOW_START, &ProtonConsts::STREAMING_WINDOW_END, &ProtonConsts::STREAMING_TIMESTAMP_ALIAS})
+                if (block_for_format.has(*reserved_col))
+                    block_for_format.erase(*reserved_col);
```

Comment:
These columns has been removed in `ProxyStream`. I think we don't need this code.